Core/SqlGeneration/SqlCommandBuilder.cs

```diff
@@ -103,6 +104,21 @@ public void AppendFormat (string stringToAppend, params object[] parameters)
       _stringBuilder.AppendFormat (stringToAppend, parameters);
     }
 
+    public void AppendCollection (ConstantCollectionExpression collectionExpression)
+    {
+      ArgumentUtility.CheckNotNull ("collectionExpression", collectionExpression);
+
+      if (collectionExpression.IsEmptyCollection)
+      {
+        Append ("SELECT NULL WHERE 1 = 0");
+      }
+      else
+      {
+        var parameter = GetOrCreateParameter (collectionExpression.ConstantExpression);
+        Append ($"SELECT [Value] FROM {parameter.Name}");
+      }
+    }
```

Comment:
I remember that we talked about that, but I thought we decided differently.
`SqlCommandBuilder` works just fine, it just creates a single parameter with a collection value. This class is MSSQL-specific, so the caller has access to the "table-valued parameters" feature to deal with collection parameter values.
The Linq.SqlBackend is used to create an `SqlCommandData` object that contains the `CommandText` and `Parameters`; it is the caller's job to create an actual `SqlCommand` from that, and he can do that because he is using MSSQL.
Using a single parameter with a collection value is not a `Remotion.Data`-specific behavior, so there is no reason to put that into `Remotion.Data`.
If we want to give the caller control over how SQL generation works for collection parameters, we should use options in `QueryExecutor` that control which `ISqlCommandBuilder` implementation is instantiated and how; right now, new SqlCommandBuilder() is hard-coded in `GenerateSqlCommand(QueryModel queryModel)`. But that would be a different task entirely.

Reply:
Ah, yes, it's SQL Server specific, but re-linq SqlBackend should not make assumptions about which custom types the library may depend on. Said differently: re-linq SqlBackend itself should only assume native SQL Server types but not require the user to write custom types beyond actual tables.

Reply:
> If we want to give the caller control over how SQL generation works for collection parameters, we should use options in QueryExecutor that control which ISqlCommandBuilder implementation is instantiated and how; right now, new SqlCommandBuilder() is hard-coded in GenerateSqlCommand(QueryModel queryModel). But changing that would be a different task entirely.

I see. no out-of-the-box answer right now. We definitely need to have an extensionpoint here, somewhere. I thought injecting the SqlCommandBuuilder was the design we desiced upon?

Reply:
As discussed: we build this as an exptension point but don't do any oppionated designs or prerequites in SqlBackend.